lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -67,7 +67,10 @@ def configure(conf)
       if @cache_ttl < 0
         @cache_ttl = :none
       end
-      @cache                                  = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      @cache = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      if @include_namespace_id
+        @namespace_cache = LruRedux::TTL::ThreadSafeCache.new([25, @cache_size].min, @cache_ttl)
```

Comment:
I just got rid of the min setting -- even if there are enough namespaces around to fill the cache, it's nothing to worry about since entry each is just a pair of short strings.